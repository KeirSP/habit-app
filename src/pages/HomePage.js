import React, { Component } from 'react';
import SignUp from '../components/SignUp';
import Login from '../components/Login';
import './homePage.css'

class HomePage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            status: ""
        }
    }
    
    handleChange = (value) => {
        return this.setState({status: value})
    }

    render() {
        if(this.state.status === "login"){
            return (
                <div >
                    <div>
                        <Login data-test="login-component"/>
                        <button className= "signup" onClick={() => this.handleChange("signup")} data-test="login-button">
                            Sign Up
                        </button>
                    </div>
                </div>
            )
        } else if (this.state.status === "signup"){
            return (
                <div>
                    <div>
                        <SignUp/>
                        <button className= "login" onClick={() => this.handleChange("login")} data-test="signUp-button">
                            Log In
                        </button>
                    </div>
                </div>
            )
        }else{
            return (
                <div className="Page" data-test="component-HomePage">
                    <h1>Welcome to Habyte</h1>
                    <h3>the habit tracker for you</h3>
                    <div>
                        <button className= "login" onClick={() => this.handleChange("login")} data-test="login-button" >
                            Log In 
                        </button>
                        <button className= "signup" onClick={() => this.handleChange("signup")} data-test="signUp-button">
                            Sign Up
                        </button>
                    </div>
                </div>
            )
        };
    }
}

export default HomePage;