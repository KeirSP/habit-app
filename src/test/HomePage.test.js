import React from 'react';
import ReactDOM from 'react-dom'
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import HomePage from '../pages/HomePage'


/* **** Basic setup ***** */
let wrapper;
const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }
const setup = (props={}, state='') => {
    const wrapper = shallow(<HomePage {...props} />)
    if (state) wrapper.setState(state);
    return wrapper;
}
/* *********************** */

describe(HomePage, () => {
    it('mounts to DOM', () => {
        const div = document.createElement('div');
        ReactDOM.render(<HomePage />, div);
        ReactDOM.unmountComponentAtNode(div);
    });


    it('Renders correctly', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    })

    it('renders without Error', () => {
        const wrapper = setup();
        const homePageComponent = findByTestAttr(wrapper, 'component-HomePage')
        expect(homePageComponent.length).toBe(1);
     })

    it('should render a <div />', () => {
        const wrapper = setup();
        expect(wrapper.find('div').length).toEqual(2);
    });

    it('should render a <button />', () => {
        const wrapper = setup();
        expect(wrapper.find('button').length).toEqual(2);
    });

    it('should render Login button',()=>{
        const wrapper = setup();
        const buttonLogin = findByTestAttr(wrapper, 'login-button')
        expect(buttonLogin.length).toBe(1);
    });

    it('should render SignUp button',()=>{
        const wrapper = setup();
        const buttonSignUp = findByTestAttr(wrapper, 'signUp-button')
        expect(buttonSignUp.length).toBe(1);
    });

    it('Status starts as an empy state', () => {
        const wrapper = setup();
        const initialLoginState = wrapper.state('status');
        expect(initialLoginState).toBe("")
    });

    it('should render SignUp component after signUp button clicked',()=>{
       const status = 'signup';
       const wrapper = setup('', {status});
       const buttonSignUp = findByTestAttr(wrapper, 'signUp-button');
       buttonSignUp.simulate('click')
       const initialLoginState = wrapper.state('status');
       expect(initialLoginState).toBe("login")
    });

    it('should render Login component after login button clicked',()=>{
        const status = 'login';
        const wrapper = setup('', {status});
        const buttonLogin = findByTestAttr(wrapper, 'login-button');
        buttonLogin.simulate('click')
        const initialLoginState = wrapper.state('status');
        expect(initialLoginState).toBe("signup")
     })

     it('should render Login component after login button clicked',()=>{
        const status = '';
        const wrapper = setup('', {status});
        const buttonLogin = findByTestAttr(wrapper, 'login-button');
        buttonLogin.simulate('click')
        const initialLoginState = wrapper.state('status');
        expect(initialLoginState).toBe("login")
     })

     it('should render SignUp component after signUp button clicked',()=>{
        const status = '';
        const wrapper = setup('', {status});
        const buttonSignUp = findByTestAttr(wrapper, 'signUp-button');
        buttonSignUp.simulate('click')
        const initialLoginState = wrapper.state('status');
        expect(initialLoginState).toBe("signup")
     });

 })


