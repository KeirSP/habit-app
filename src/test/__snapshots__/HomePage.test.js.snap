// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Renders correctly 1`] = `ShallowWrapper {}`;
